listenbrainz/webserver/views/api_tools.py

```diff
@@ -114,42 +114,43 @@ def validate_listen(listen, listen_type):
 
     if listen_type in (LISTEN_TYPE_SINGLE, LISTEN_TYPE_IMPORT):
         if 'listened_at' not in listen:
-            log_raise_400("JSON document must contain the key listened_at at the top level.", listen)
+            raise APIBadRequest("JSON document must contain the key listened_at at the top level.", listen)
 
         try:
             listen['listened_at'] = int(listen['listened_at'])
         except ValueError:
-            log_raise_400("JSON document must contain an int value for listened_at.", listen)
+            raise APIBadRequest("JSON document must contain an int value for listened_at.", listen)
 
         if 'listened_at' in listen and 'track_metadata' in listen and len(listen) > 2:
-            log_raise_400("JSON document may only contain listened_at and "
-                           "track_metadata top level keys", listen)
+            raise APIBadRequest("JSON document may only contain listened_at and "
+                                "track_metadata top level keys", listen)
 
         # if timestamp is too high, raise BadRequest
         # in order to make up for possible clock skew, we allow
         # timestamps to be one hour ahead of server time
         if not is_valid_timestamp(listen['listened_at']):
-            log_raise_400("Value for key listened_at is too high.", listen)
+            raise APIBadRequest("Value for key listened_at is too high.", listen)
 
     elif listen_type == LISTEN_TYPE_PLAYING_NOW:
-        if 'listened_at' in listen:
-            log_raise_400("JSON document must not contain listened_at while submitting "
-                           "playing_now.", listen)
+        if listen is not None:
+            if 'listened_at' in listen:
+                raise APIBadRequest("JSON document must not contain listened_at while submitting "
+                                    "playing_now.", listen)
 
-        if 'track_metadata' in listen and len(listen) > 1:
-            log_raise_400("JSON document may only contain track_metadata as top level "
-                           "key when submitting now_playing.", listen)
+            if 'track_metadata' in listen and len(listen) > 1:
+                raise APIBadRequest("JSON document may only contain track_metadata as top level "
+                                    "key when submitting now_playing.", listen)
```

Comment:
can you open a ticket for this or fix to be consistent with the docs - one error message says now_playing and the next one down says playing now